### PR TITLE
feat: go multipart

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

Adding multipart support for go

## Test Plan

### CODE REVIEW

File: templates/go/services/service.go.twig [hunk 1]
```diff
@@ -95,7 +99,7 @@ func (srv *{{ service.name | caseUcfirst }}) {{ method.name | caseUcfirst }}({{
 	path := "{{ method.path }}"
 {% endif %}
 {{include('go/base/params.twig')}}
-{% if 'multipart/form-data' in method.consumes %}
+{% if 'multipart/form-data' in method.consumes and method.name | lower != "createexecution" %}
```
Comment (christyjacob4) [inline templates/go/services/service.go.twig:102]: I think we discussed to use method.type or so here

File: templates/go/base/params.twig [hunk 2]
```diff
@@ -10,7 +10,11 @@
 	params["{{ parameter.name }}"] = {{ parameter.name | caseUcfirst }}
 {% else %}
 	if options.enabledSetters["{{ parameter.name | caseUcfirst}}"] {
+	{%~ if method.name | caseLower == "createexecution" and parameter.name == "body" %}
+		params["{{ parameter.name }}"] = string(options.{{ parameter.name | caseUcfirst }}.Data)
```
Comment (christyjacob4) [inline templates/go/base/params.twig:14]: this again feels too specific to createexecution... 

why dont we copy over the stuff from php ?

File: templates/go/services/service.go.twig (hunk 1 above)
Comment (byawitz) [reply templates/go/services/service.go.twig:102]: This is to make sure it's not the `createexecution` method. the `method.type` won't cover all use-cases

File: templates/go/services/service.go.twig [hunk 3]
```diff
@@ -1,12 +1,16 @@
 {%- set requireModelsPkg = false -%}
-{%- set requireFilesPkg = false -%}
+{%- set requirePayloadPkg = false -%}
+{%- set requireParsingLibes = false -%}
```
Comment (christyjacob4) [inline templates/go/services/service.go.twig:3]: where is this var used ? And what is it used for?

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**